### PR TITLE
naming_convention: Add space-case algorithm

### PR DESCRIPTION
**Pull request by Asnelchristian**

### CODE REVIEW

File: coalib/bearlib/naming_conventions/__init__.py [hunk 1]
```diff
@@ -74,3 +83,36 @@ def to_snakecase(string):
...
+   Converts the given string to space-case.
+
+   >>> to_spacecase('helloWorld')
+   'Hello World'
+   >>> to_spacecase('__Init__File__')
+   'Init File'
+   >>> to_spacecase('')
+   ''
+   >>> to_spacecase('Already Space Case')
+   'Already Space Case'
```
Comment (hypothesist) [inline coalib/bearlib/naming_conventions/__init__.py]: @Asnelchristian what is the expected outpuot for a string like `ABCD`?

`A B C D` or `ABCD` itself?

Comment (Asnelchristian) [inline coalib/bearlib/naming_conventions/__init__.py]: `A B C D`

File: coalib/bearlib/naming_conventions/__init__.py [hunk 2]
```diff
@@ -17,6 +17,9 @@ def to_camelcase(string):
     :param string: The string to convert.
     :return:       The camel-cased string.
     """
+    string = re.sub("(\s)",
+                    lambda match: '_',
+                    string)
```
Comment (sils1297) [inline coalib/bearlib/naming_conventions/__init__.py]: why do you introduce those lines? They don't add the new algorithm so they don't belong into this commit right? (You can use `git reset master` and `git add -p` to make new commits out of the parts.)

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2381/commits/f9b6e52b0e43158018beefc7b58015497773fe58

**Comment by sils1297** [comment]
also test coverage seems to drop, I hate codecov for not failing this build

**Comment by Asnelchristian** [comment]
why is the coverage dropping ?

**Comment by sils1297** [comment]
I'd guess because you add a function that is untested?

**Comment by Asnelchristian** [comment]
i see i will add doctests then

**Comment by Asnelchristian** [comment]
It is now perfect :)

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2381/commits/a883a5cd9007c2a5693d5cc0422ec45f9f4ff5bf https://github.com/coala-analyzer/coala/pull/2381/commits/aef8ec970af021397e8b13feda26574c87b6f448

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2381/commits/78b3df4991d484041632d19d38a647430738c47b

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**